Rework the MQTT analyzer into a plugin and bro-pkg

sethhall opened the pull request:
I restructured it all to be a plugin external to Bro and applied the BSD license as we discussed separately.

Commit:
Turn the repository into a bro-pkg

Commit:
Adding a .gitignore.

Commit:
Clean up the readme on disk

Commit:
Add a readme symlink with an extension

Commit:
Get rid of messages dumping to stdout.

sethhall commented:
Oh, you may want to rename this repository like I named mine too (bro-mqtt)?  It will make sense to break analyzers out into multiple repositories if you write multiple of them.

supriyask commented:
Hi Seth,

Thanks☺️. Yes I will change the name of the repository. Also, it's been
quite sometime since I wrote that code. I am not able to recall but there
was some issue, that's the reason I couldn't take it forward. I will dig in
my notes and emails and let you know.

I will try to email you the details by tomorrow or else Monday (I am moving
to a new place, so busy this weekend).

You have a great day.

Regards,
Supriya

On Sat, Apr 28, 2018 at 12:59 AM Seth Hall <notifications@github.com> wrote:

> Oh, you may want to rename this repository like I named mine too
> (bro-mqtt)? It will make sense to break analyzers out into multiple
> repositories if you write multiple of them.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/supriyask/Bro/pull/2#issuecomment-385150854>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AKB091GBV7BCABAc6XHgARugMr6fkYKWks5ttCFugaJpZM4TrTJD>
> .
>
-- 
Sent from my iPhone

Commit:
Lots more changes.

 - Restructured analyzer to make it more readable.
   The commands are all implemented as case statement
   extensions.
 - Parse remaining_length correctly as a variably sized int.
 - Lots of clean up to the events to pass data into them
   correctly so that more accurate scripts can be written.
 - Passed the payload field for the publish message into
   the publish event.
 - Tons of little bug fixes.

Commit:
Implement some final DPD protocol handling.
